articles/tutorials/advanced/2d_shaders/06_color_swap_effect/index.md

```diff
@@ -0,0 +1,643 @@
+    Texture = <ColorMap>;
+    MinFilter = Point;
+    MagFilter = Point;
+    MipFilter = Point;
+    AddressU = Clamp;
+    AddressV = Clamp;
+};
+```
+
+The `Texture2D` and `sampler2D` declarations are required to read from textures in a MonoGame shader. A `Texture2D` represents the pixel data of the image. A `sampler2D` defines _how_ the shader is allowed to read data from the `Texture2D`. 
+
+The `ColorMapSampler` has a lot of extra properties (`MinFilter`, `MagFilter`, `MipFilter`, `AddressU`, and `AddressV`) that control exactly how the texture data is read from the `ColorMap`. 
+
+By default, when a sampler reads data from a texture in a shader, it will subtly blend nearby pixel values to increase the visual quality. However, when a texture is being used as a lookup table, this blending is problematic because it will distort the data stored in the texture. The `Point` value given to the `Filter` properties tells the sampler to only read _one_ pixel value. 
+
+When a sampler is reading a texture, there is always some _location_ being used to read pixel data from. The texture coordinate space is from 0 to 1, in both the `u` and `v` axes. By default, if a value greater than 1, or less than 0 is given, the sampler will _wrap_ the value around to be within the range 0 to 1. For example, `1.15` would become `0.15`.  The `Clamp` value prevents the wrapping and cuts the input off at the min and max values. For example, `1.15` becomes `1.0`. 
+
+> [!tip] 
+> More information on Samplers.
+> 
+> The [MonoGame Docs](https://docs.monogame.net/articles/getting_to_know/whatis/graphics/WhatIs_Sampler.html) have more details on samplers. 
+
+The shader function can now do 2 steps to perform the color swap,
+1. read the original color value of the pixel,
+2. use the original color's red value as the `key` in the lookup texture to extract the swap color `value`. 
+
+To help visualize the effect, it will be helpful to visualize the original color _and_ the swap color. Add a control parameter that can be used to select between the two colors:
+
+```hlsl
+// a control variable to lerp between original color and swapped color  
+float OriginalAmount;
+```
+
+Change the shader function to the following:
+
+```hlsl
+float4 MainPS(VertexShaderOutput input) : COLOR
+{
+    // read the original color value
+    float4 originalColor = tex2D(SpriteTextureSampler,input.TextureCoordinates);
+    
+    // produce the key location
+    //  note the x-offset by half a texel solves rounding errors.
+    float2 keyUv = float2(originalColor.r;
+    
+    // read the swap color value
+    float4 swappedColor = tex2D(ColorMapSampler, keyUv) * originalColor.a;
+    
+    // return the result color
+    return lerp(swappedColor, originalColor, OriginalAmount);
+}
+```
+
+Now in the game, we can visualize the color swap by adjusting the control parameter. Perhaps the colors we picked do not look very nice.
+![Figure 6.8: The color swap effect is working!](./gifs/color-swap.gif)
```

Comment:
Another change in the IMGUI screen, hopefully this will be explained earlier in the chapter.